filament/src/driver/opengl/OpenGLDriver.cpp

```diff
@@ -2327,12 +2327,23 @@ void OpenGLDriver::setRenderPrimitiveBuffer(Driver::RenderPrimitiveHandle rph,
         GLVertexBuffer const* const eb = handle_cast<const GLVertexBuffer*>(vbh);
         GLIndexBuffer const* const ib = handle_cast<const GLIndexBuffer*>(ibh);
 
-        assert(ib->elementSize == 2 || ib->elementSize == 4);
+        assert(ib->elementSize == 1 || ib->elementSize == 2 || ib->elementSize == 4);
 
         bindVertexArray(rp);
         CHECK_GL_ERROR(utils::slog.e)
 
-        rp->gl.indicesType = ib->elementSize == 4 ? GL_UNSIGNED_INT : GL_UNSIGNED_SHORT;
+        switch (ib->elementSize) {
```

Comment:
If we're going to do this, we need to update the Vulkan and Metal backends too.

I'm not sure this is worth it though because I'm told no hardware actually supports 8-bits anyways -- the driver (or backend) will have to massage the data.

Reply:
For this to go in we'll need to add proper support for Metal and Vulkan indeed.

Reply:
Metal does not natively support 8 bit indices.